source/s3_auto_ranged_get.c

```diff
@@ -10,83 +16,39 @@
-    AWS_S3_AUTO_RANGED_GET_STATE_ALL_REQUESTS,
-    AWS_S3_AUTO_RANGED_GET_STATE_ALL_REQUESTS_MADE
-};
-
-enum aws_s3_auto_ranged_get_request_type {
-    AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_FIRST_PART,
-    AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_PART
-};
-
-struct aws_s3_auto_ranged_get {
-    struct aws_s3_meta_request base;
-
-    struct {
-        enum aws_s3_auto_ranged_get_state state;
-
-        uint32_t next_part_number;
-        uint32_t total_num_parts;
-        uint32_t num_parts_completed;
-        size_t total_object_size;
-
-    } synced_data;
-};
-
-static void s_s3_auto_ranged_get_lock_synced_data(struct aws_s3_auto_ranged_get *auto_ranged_get);
-static void s_s3_auto_ranged_get_unlock_synced_data(struct aws_s3_auto_ranged_get *auto_ranged_get);
+const uint32_t s_conservative_max_requests_in_flight = 8;
 
 static void s_s3_meta_request_auto_ranged_get_destroy(struct aws_s3_meta_request *meta_request);
 
-static int s_s3_auto_ranged_get_next_request(
+static void s_s3_auto_ranged_get_update(
     struct aws_s3_meta_request *meta_request,
-    struct aws_s3_request **out_request);
+    uint32_t flags,
+    struct aws_s3_request **out_request,
+    enum aws_s3_meta_request_update_status *out_status);
 
 static int s_s3_auto_ranged_get_prepare_request(
     struct aws_s3_meta_request *meta_request,
     struct aws_s3_client *client,
     struct aws_s3_vip_connection *vip_connection,
     bool is_initial_prepare);
 
-static int s_s3_auto_ranged_get_header_block_done(
-    struct aws_http_stream *stream,
-    enum aws_http_header_block header_block,
-    struct aws_s3_vip_connection *vip_connection);
-
-static void s_s3_auto_ranged_get_notify_request_destroyed(
+static void s_s3_auto_ranged_get_request_finished(
     struct aws_s3_meta_request *meta_request,
-    struct aws_s3_request *request);
+    struct aws_s3_request *request,
+    int error_code);
 
 static struct aws_s3_meta_request_vtable s_s3_auto_ranged_get_vtable = {
-    .next_request = s_s3_auto_ranged_get_next_request,
+    .update = s_s3_auto_ranged_get_update,
     .send_request_finish = aws_s3_meta_request_send_request_finish_default,
     .prepare_request = s_s3_auto_ranged_get_prepare_request,
     .init_signing_date_time = aws_s3_meta_request_init_signing_date_time_default,
     .sign_request = aws_s3_meta_request_sign_request_default,
-    .incoming_headers = NULL,
-    .incoming_headers_block_done = s_s3_auto_ranged_get_header_block_done,
-    .incoming_body = NULL,
-    .stream_complete = NULL,
-    .notify_request_destroyed = s_s3_auto_ranged_get_notify_request_destroyed,
+    .finished_request = s_s3_auto_ranged_get_request_finished,
+    .delivered_requests = aws_s3_meta_request_delivered_requests_default,
     .destroy = s_s3_meta_request_auto_ranged_get_destroy,
     .finish = aws_s3_meta_request_finish_default,
 };
 
-static void s_s3_auto_ranged_get_lock_synced_data(struct aws_s3_auto_ranged_get *auto_ranged_get) {
```

Comment:
This was locking the base mutex anyway, so now just using the base type lock/unlock functions.